src/Pate/Equivalence.hs

```diff
@@ -397,34 +331,66 @@ positiveMemCells ::
   W4.IsSymExprBuilder sym =>
   MM.RegisterInfo (MM.ArchReg arch) =>
   sym ->
+  MemRegionEquality sym arch ->
   SimState sym arch PBi.Original ->
   SimState sym arch PBi.Patched ->
-  MemEquivRelation sym arch ->
   PMC.MemCellPred sym arch ->
   IO (W4.Pred sym)
-positiveMemCells sym stO stP memEq memPred = do
+positiveMemCells sym memEqRegion stO stP memPred = do
   let
     memCells = PMC.predToList memPred
-    resolveCellPos = resolveCellEquiv sym stO stP memEq
+    resolveCellPos = resolveCellEquiv sym memEqRegion stO stP
   foldr (\(Some cell, cond) -> andM sym (resolveCellPos cell cond)) (return $ W4.truePred sym) memCells
 
 resolveCellEquiv ::
```

Comment:
Is this generally stack-specific, or just because the next two functions use it for stack reasoning?

Reply:
It's stack specific because that's the only region we're treating specially, but I'll rename it here since this function doesn't really care what the region is meant to represent.